### ISSUE TITLE
ASTRAL-CHAIN performance issue in latest EA built

### ISSUE THREAD

**Issue by bccyben** [yuzu-emu/yuzu#7277]
Use yuzu EA 1887 the emulation works fine.
But with the latest EA build like 2177 even the latest mainline build which is 795, the game will be very stuttering.

I am using an Nvidia rtx 2080 super GPU and an overclocked i5 9600k CPU.
I am sure I enable the GPU Accelerated ASTC Texture Decoding, I also have tried to disable it and the stutter in the game become worse.

**Comment by G-Spawn** [comment]
Please provide a log file when you can like new issue template asks for

**Comment by bccyben** [comment]
> Please provide a log file when you can like new issue template asks for

Apologize for I forget the log file, now it is uploaded.

**Comment by bccyben** [comment]
https://user-images.githubusercontent.com/32619018/140012118-582c6dad-a7ee-4fc2-b0e3-4ac0c1d552b9.mp4
Here is the video file, you can see it have some stutters in game.

**Comment by G-Spawn** [comment]
The potential reason is usage of latest drivers and async shaders. Please try nvidia driver 472.12 and compare.

**Comment by bccyben** [comment]
> The potential reason is usage of latest drivers and async shaders. Please try nvidia driver 472.12 and compare.

Thank you for your replay.
I have tried nvidia driver 472.12 and the the game still very stuttering.
I noticed when the game stuttering, console shows "Service.NVDRV <Warning> core\hle\service\nvdrv\devices\nvhost_ctrl.cpp:IocCtrlClearEventWait:209: cleared event wait on, event_id: 0"

You can confirm it in log file
[yuzu_log.txt.old.txt](https://github.com/yuzu-emu/yuzu/files/7468011/yuzu_log.txt.old.txt)
.

**Comment by G-Spawn** [comment]
try disabling async shaders and perhaps compare in vulkan.

**Comment by FernandoS27** [comment]
Best is that you try Vulkan. For many games it's better than OGL, even on NVIDIA.

**Comment by FernandoS27** [comment]
could you help us find which one caused the regression: https://github.com/yuzu-emu/yuzu-mainline/releases

you don't have to go 1 by 1. just go back 50 or so versions and see if it works. If it does, try somewhere in the middle. Kind like loking in a dictionary.

**Comment by bccyben** [comment]
> could you help us find which one caused the regression: https://github.com/yuzu-emu/yuzu-mainline/releases
> 
> you don't have to go 1 by 1. just go back 50 or so versions and see if it works. If it does, try somewhere in the middle. Kind like loking in a dictionary.

I have done the test, here is the result:

bad
736
735

good
717
726
730
733
734

So I guess mainline build 734 was the latest version do not have this issue.
I hope my work can help yuzu team to fix this issue.

735:
https://github.com/yuzu-emu/yuzu-mainline/releases/tag/mainline-0-735

734:
https://github.com/yuzu-emu/yuzu-mainline/releases/tag/mainline-0-734

**Comment by G-Spawn** [comment]
your dillegance in reporting is appreciated. thank you kindly! Hopefully that helps narrow down what potential change(s) might have caused issue.

**Comment by Iridescent-me** [comment]
@bccyben @G-Spaw

https://user-images.githubusercontent.com/75098518/144741327-201695ff-50bc-4a7c-95c3-3ee7b20f8c37.mp4

This issue still exists...If you disable "GPU cache garbage collection" will be smooth, but now it is enabled by default, so can it be disabled in ASTRAL CHAIN?

**Comment by AkumaBugs** [comment]
Hi, Getting the same stuttering in the latest version of yuzu. Are any other games having issues of sudden stutters which were previously absent?

**Comment by DuMOHsmol** [comment]
Can confirm it as well. The game seems to use a lot of VRAM and garbage collection is causing a lot of stuttering. Commenting out the code inside `TextureCache<P>::RunGarbageCollector()` resolves the problem for me.

**Comment by retrofan2019** [comment]
yea I can confirm that it works fine with older versions that had the option to enable/disable "GPU cache garbage collection" ..but with newer versions it just stutters to the point that it's completely unplayable .
it would be great if the option is restored that way we can disable it for this game or any other game that has a similar problem.

**Comment by sayanss2** [comment]
The value of `minimum_ memory` (texture_cache.h) is incorrectly selected. It is probably more correct to run Garbage Collector to check `expected_memory`:

` if (total_used_memory > expected_memory) {
        RunGarbageCollector();
    }`

The value of the `minimum_ memory` variable is currently **0** (is this correct?), which in fact constantly calls Garbage Collector.
If you limit the GC call relative to `expected_memory`, then the problem is solved on cards with a capacious video buffer (tested on 3070 c 8GB VRAM).

For example, during tests I got the following memory values in debugging:
`total_used_memory`=2977803264
`expected_memory`=3373898137
`critical_memory`=5904321740
That is, the value of `total_used_memory` on the example of the ASTRAL CHAIN game did not exceed the value of `expected_memory`.

**Comment by Iridescent-me** [comment]
Hello, I've been paying attention to this issues and hope it can be solved. I see that GC is optimized in pull #7720 , but in the latest version, turning the camera is still very stuttering.
My GPU is 1650 4G. Is it because I have too little VRAM?
Can you temporarily return the GC option until it is solved perfectly? It can be enabled by default, but let us have the option to disable it. @FernandoS27 @sayanss2

**Comment by FernandoS27** [comment]
> Hello, I've been paying attention to this issues and hope it can be solved. I see that GC is optimized in pull #7720 , but in the latest version, turning the camera is still very stuttering. My GPU is 1650 4G. Is it because I have too little VRAM? Can you temporarily return the GC option until it is solved perfectly? It can be enabled by default, but let us have the option to disable it. @FernandoS27 @sayanss2

are you rescaling? If so, you may not be able to afford it for that game. That game uses a lot of RAM due to ASTC textures being transformed. I'm against an option as that gives a hell of issues for the support team. That option easily can cause a lot of random crashes and trust me, people just keep marking things on/off without looking that they break things. We have such issue with Async Shaders already.

**Comment by Jorge9811** [comment]
> people just keep marking things on/off without looking that they break things

could make it so it can only be toggled from a ini or alternatively put it in the gui behind a checkbox that explicitly says that no support is given if enabled kina like a experimental options warning or some thing

**Comment by retrofan2019** [comment]
> 

I was about to say the same thing ... it could be one of the options that can be enabled/disabled only through ini file ... that way only people that know exactly what they r doing can use it.

**Comment by Iridescent-me** [comment]
> are you rescaling? If so, you may not be able to afford it for that game. That game uses a lot of RAM due to ASTC textures being transformed. I'm against an option as that gives a hell of issues for the support team. That option easily can cause a lot of random crashes and trust me, people just keep marking things on/off without looking that they break things. We have such issue with Async Shaders already.

Oh, sorry.I didn't know it would crash the game. Maybe as mentioned above... it can only be set in ini.

**Comment by chikcne** [comment]
Any update on #7720? It seems to be almost complete barring a few nitpicks but maybe the changes aren't working as expected?

**Comment by bzlom** [comment]
Build 994 - issue with stuttering still happening both with Vulcan and OpenGL for Astral Chain v1.0.0 and v1.0.1. 

PC specs: 
CPU: i5-10600KF
RAM: 16GB Patriot Viper 4 Blackout Series DDR4 3600MHz
GPU: Geforce 2060 (Driver version 511.79)
SSD: Samsung EVO 860

**Comment by gidoly** [comment]
is still happens?

**Comment by AkumaBugs** [comment]
Stuttering still happens. Checked on EA 2888 version. Playing this game on Mainline 734 to get ridd of stuttering.
